testsuite/MDAnalysisTests/core/test_groups.py

```diff
@@ -522,3 +522,42 @@ def test_groupby_int(self):
 
         for g in gb.values():
             assert_(len(g) == 5)
+
+
+class TestAttributeSetting(object):
+    def setUp(self):
+        pass
+
+    def tearDown(self):
+        pass
+
+    @staticmethod
+    def _check_setattr_fails(thing):
+        assert_raises(AttributeError, setattr, thing, 'this', 'that')
+
+    def test_setting_fail(self):
+        u = make_Universe()
+
+        ag, rg, sg = u.atoms[:10], u.residues[:5], u.segments[:3]
+        a, r, s = u.atoms[10], u.residues[6], u.segments[3]
+
+        for thing in [ag, rg, sg, a, r, s]:
+            yield self._check_setattr_fails, thing
+
+    def test_making_subclass_works(self):
+        class AwesomeGroup(groups.AtomGroup):
+            # I want to create a subclass of AtomGroup
+            # that has a new attribute, should be possible..
+            def __init__(self, ix, u, thing):
+                super(AwesomeGroup, self).__init__(ix, u)
```

Comment:
Random attribute should work. We could try and make the whitelist extensible, but I'd rather just remove the safety rail in subclasses